![Callysto.ca Banner](https://github.com/callysto/curriculum-notebooks/blob/master/callysto-notebook-banner-top.jpg?raw=true)

# Callysto’s Weekly Data Visualization

## Solar Energy

### Recommended Grade levels: 5-7
<br>

### Instructions

Click "Cell" and select "Run All".

This will import the data and run all the code, so you can see this week's data visualization. Scroll back to the top after you’ve run the cells.

![instructions](https://github.com/callysto/data-viz-of-the-week/blob/main/images/instructions.png?raw=true)

**You don't need to do any coding to view the visualizations**.

The plots generated in this notebook are interactive. You can hover over and click on elements to see more information. 

Email contact@callysto.ca if you experience issues.

### About this Notebook

Callysto's Weekly Data Visualization is a learning resource that aims to develop data literacy skills. We provide Grades 5-12 teachers and students with a data visualization, like a graph, to interpret. This companion resource walks learners through how the data visualization is created and interpreted by a data scientist. 

The steps of the data analysis process are listed below and applied to each weekly topic.

1. Question - What are we trying to answer?
2. Gather - Find the data source(s) you will need. 
3. Organize - Arrange the data, so that you can easily explore it. 
4. Explore - Examine the data to look for evidence to answer the question. This includes creating visualizations. 
5. Interpret - Describe what's happening in the data visualization. 
6. Communicate - Explain how the evidence answers the question. 

# Question

Is solar energy, a renewable energy source, a viable energy source today? What benefits does it offer compared to other common sources of energy used today? 

Solar energy in simple terms is the energy we get from the sun's light and heat. The capturing of energy is made possible using specialized tools, such as solar panels. One of the standout benefits of solar energy is that its considered a renewable energy source. As long as the sun continues to shine, we can keep producing energy, ensuring a constant and sustainable power supply. 

Solar energy is also exceptionally eco-friendly. Unlike traditional energy generation, solar technologies operate without emitting any harmful pollutants into the air. This contrasts with many other energy sources, such as fossil fuels, which are finite and can release pollutants into the environment. This means that using solar energy helps keep our air clean and reduces our carbon footprint, contributing to a healthier planet.

Due to these significant advantages, many cities, including Calgary, are increasingly embracing solar energy as a key component of their energy mix. Now, let's delve into Calgary's specific solar energy initiatives to understand just how beneficial this renewable source is compared to traditional energy sources.

### Goal

Our goal is to look at the production of energy by different solar energy stations around Calgary and discover trends, such as identifying certain months that output more energy than others. Based on this information, we can make informed decisions about optimizing energy production and distribution, enabling more efficient utilization of solar resources to meet the energy demands of Calgary's residents.

The dataset used in this notebook is taken from [Calgary's Open Data Portal](https://data.calgary.ca/Environment/Solar-Energy-Production/ytdn-2qsp), specifically their *solar energy production* section.

# Gather

### Code:

Run the code cells below to import the libraries we need for this project. Libraries are pre-made code that make it easier to analyze our data.

In [35]:
import pandas as pd
import plotly.express as px
from plotly.subplots import make_subplots
import plotly.graph_objects as go
import folium
print("Libaries imported")

Libaries imported


### Data

This notebook utilizes solar energy production data provided by the city of *Calgary*. Specifically, it contains the hourly output at each of Calgary's solar photovoltaic projects. For additional information regarding each site, see [this link]([calgary.ca/solar](https://data.calgary.ca/dataset/City-of-Calgary-Solar-Photovoltaic-Sites/vrdj-ycb5)).

### Import the data

In [36]:
# Import the data
solar_data = pd.read_csv("https://raw.githubusercontent.com/callysto/data-files/main/data-viz-of-the-week/solar-energy/solar_energy.csv")
solar_data

name      id                   address  \
0       Calgary Fire Hall Headquarters  314106             1212 42 AV SE   
1       Calgary Fire Hall Headquarters  314106             1212 42 AV SE   
2       Calgary Fire Hall Headquarters  314106             1212 42 AV SE   
3       Calgary Fire Hall Headquarters  314106             1212 42 AV SE   
4       Calgary Fire Hall Headquarters  314106             1212 42 AV SE   
...                                ...     ...                       ...   
282462  Bearspaw Water Treatment Plant  577650  11444 Bearspaw Dam RD NW   
282463  Bearspaw Water Treatment Plant  577650  11444 Bearspaw Dam RD NW   
282464  Bearspaw Water Treatment Plant  577650  11444 Bearspaw Dam RD NW   
282465  Bearspaw Water Treatment Plant  577650  11444 Bearspaw Dam RD NW   
282466  Bearspaw Water Treatment Plant  577650  11444 Bearspaw Dam RD NW   

                          date      kWh  \
0       2017/09/11 08:00:00 AM    1.130   
1       2017/09/11 09:00:00 AM    2.340   
2       2017/09/11 10:00:00 AM    3.656   
3       2017/09/11 11:00:00 AM    4.577   
4       2017/09/11 12:00:00 PM    6.506   
...                        ...      ...   
282462  2023/05/06 08:00:00 PM    0.012   
282463  2023/06/02 01:00:00 PM  158.581   
282464  2023/06/02 02:00:00 PM  152.632   
282465  2023/06/02 03:00:00 PM  132.481   
282466  2023/06/02 04:00:00 PM   80.433   

                                               public_url installationDate  \
0       https://monitoring.solaredge.com/solaredge-web...       2016/11/07   
1       https://monitoring.solaredge.com/solaredge-web...       2016/11/07   
2       https://monitoring.solaredge.com/solaredge-web...       2016/11/07   
3       https://monitoring.solaredge.com/solaredge-web...       2016/11/07   
4       https://monitoring.solaredge.com/solaredge-web...       2016/11/07   
...                                                   ...              ...   
282462  https://monitoringpublic.solaredge.com/solared...       2017/10/31   
282463  https://monitoringpublic.solaredge.com/solared...       2017/10/31   
282464  https://monitoringpublic.solaredge.com/solared...       2017/10/31   
282465  https://monitoringpublic.solaredge.com/solared...       2017/10/31   
282466  https://monitoringpublic.solaredge.com/solared...       2017/10/31   

                              uid  
0       3141062017-09-11 08:00:00  
1       3141062017-09-11 09:00:00  
2       3141062017-09-11 10:00:00  
3       3141062017-09-11 11:00:00  
4       3141062017-09-11 12:00:00  
...                           ...  
282462  5776502023-05-06 20:00:00  
282463  5776502023-06-02 13:00:00  
282464  5776502023-06-02 14:00:00  
282465  5776502023-06-02 15:00:00  
282466  5776502023-06-02 16:00:00  

[282467 rows x 8 columns]

### Comment on the Data

Looking at our dataset, it appears in a *spreadsheet-like* format. That is, there are rows and columns that define the type of information that is contained in each cell. In Python (the coding language we are using), instead of referring to this structure as a spreadsheet, we call it a *dataframe*. 

To make understanding the columns in our dataframe easier, we'll be organizing the data in our next section.

# Organize

Earlier we talked about organizing our data. In coding terms, this is known as data-cleaning. Data cleaning is a crucial process in preparing and making sense of the data presented which involves identifying and correcting errors, inconsistencies, and inaccuracies in a dataset to ensure that the data is reliable for analysis. 


Beginning our data-cleaning process, let's remove information which is irrelevant to our analysis. We notice that certain columns contain irrelevant information or are undermined by repetitious information. These columns are `public_url`, `uid`, and `id`. Let's remove these columns.

In [37]:
solar_data.drop(columns=['public_url', 'uid', 'id'], inplace=True)
solar_data

name                   address  \
0       Calgary Fire Hall Headquarters             1212 42 AV SE   
1       Calgary Fire Hall Headquarters             1212 42 AV SE   
2       Calgary Fire Hall Headquarters             1212 42 AV SE   
3       Calgary Fire Hall Headquarters             1212 42 AV SE   
4       Calgary Fire Hall Headquarters             1212 42 AV SE   
...                                ...                       ...   
282462  Bearspaw Water Treatment Plant  11444 Bearspaw Dam RD NW   
282463  Bearspaw Water Treatment Plant  11444 Bearspaw Dam RD NW   
282464  Bearspaw Water Treatment Plant  11444 Bearspaw Dam RD NW   
282465  Bearspaw Water Treatment Plant  11444 Bearspaw Dam RD NW   
282466  Bearspaw Water Treatment Plant  11444 Bearspaw Dam RD NW   

                          date      kWh installationDate  
0       2017/09/11 08:00:00 AM    1.130       2016/11/07  
1       2017/09/11 09:00:00 AM    2.340       2016/11/07  
2       2017/09/11 10:00:00 AM    3.656       2016/11/07  
3       2017/09/11 11:00:00 AM    4.577       2016/11/07  
4       2017/09/11 12:00:00 PM    6.506       2016/11/07  
...                        ...      ...              ...  
282462  2023/05/06 08:00:00 PM    0.012       2017/10/31  
282463  2023/06/02 01:00:00 PM  158.581       2017/10/31  
282464  2023/06/02 02:00:00 PM  152.632       2017/10/31  
282465  2023/06/02 03:00:00 PM  132.481       2017/10/31  
282466  2023/06/02 04:00:00 PM   80.433       2017/10/31  

[282467 rows x 5 columns]

Perfect! We now have a dataframe which is more concise and easier to visualize. Next, let's identify if there are any *null* values in our dataframe. Null values are essentially placeholders that represent missing or undefined data. It's similar to having a blank space in a spreadsheet where you expected to find information. By recognizing where null values exists, we can decide how to handle them, whether it's by filling in missing information or removing incomplete entries. 

In [38]:
null_count = solar_data.isnull().sum()
print(null_count)

name                0
address             0
date                0
kWh                 0
installationDate    0
dtype: int64


Luckily, it appears that our dataframe doesn't contain any null values. We can finish our data-cleaning here, and move onto interacting with the data within our dataframe. 

# Explore

Examining the columns in our dataframe, we can observe that both the name and address columns serve to identify the individual solar energy stations across Calgary and their respective locations. To gain insight into the number of unique energy stations and their varied locations, let's proceed with analysis.

In [52]:
# Find names and addresses
names = solar_data['name'].unique()
addresses = solar_data['address'].unique()
installationDate = solar_data['installationDate'].unique()

# Print names and addresses
for i in range(len(names)):
    print(names[i], '(' + addresses[i] + ')' + ' - ' +  'Installed on: ' + installationDate[i])

Calgary Fire Hall Headquarters (1212 42 AV SE) - Installed on: 2016/11/07
Whitehorn Multi-Service Centre (3705 35 ST NE) - Installed on: 2016/12/13
Southland Leisure Centre (2000 SOUTHLAND DR SW) - Installed on: 2015/09/01
Hillhurst Sunnyside Community Association (1320 5 AV NW) - Installed on: 2016/09/25
Glenmore Water Treatment Plant (1634 56 AV SW) - Installed on: 2017/01/11
City of Calgary North Corporate Warehouse (2340 22 ST NE) - Installed on: 2016/12/21
Richmond - Knob Hill Community Hall (2433 26 AV SW) - Installed on: 2016/11/21
CFD Firehall #7 (2708 4 ST NW) - Installed on: 2017/09/13
Manchester Building M (651M 25 AV SE) - Installed on: 2017/10/23
Telus Spark (220 ST GEORGES DR NE) - Installed on: 2017/11/27
Bearspaw Water Treatment Plant (11444 Bearspaw Dam RD NW) - Installed on: 2017/10/31


Looking at the output, we see that there is a good variety of different solar stations in our dataframe. Specifically, we have 3 stations located in the NW, 3 stations located in the NE, 3 stations located in the SW, and 2 stations located in the SE. This variety in location is advantageous for several reasons.

Firstly, it promotes a more diversified and resilient energy infrastructure across Calgary. By having solar stations spread across different regions of the city, we reduce the risk of localized energy shortages or disruptions.

Secondly, it contributes to a more equitable distribution of clean energy resources, ensuring that residents from various areas have access to sustainable power sources.

Lastly, this geographical diversity allows us to study the impact of different environmental factors, such as sunlight exposure and weather patterns, on energy production, helping us optimize the efficiency of these stations and make informed decisions about future solar energy projects in Calgary.

Now that we know the number of different solar stations that are located in our dataframe, let's find how many different data-entries we have of each solar station. 

In [40]:
counts_of_names = solar_data.name.value_counts()
counts_of_names

Southland Leisure Centre                     34168
Whitehorn Multi-Service Centre               30312
City of Calgary North Corporate Warehouse    29681
Richmond - Knob Hill Community Hall          29169
Glenmore Water Treatment Plant               28266
Calgary Fire Hall Headquarters               28193
Bearspaw Water Treatment Plant               27435
CFD Firehall #7                              25231
Manchester Building M                        25117
Hillhurst Sunnyside Community Association    24578
Telus Spark                                    317
Name: name, dtype: int64

It appears that *Southland Leisure Centre* has the highest number of data entries, with 34,168 records, with *Whitehorn Multi-Service Centre* following closely with 30,312 records. *Telus Spark* has the fewest entries, with just 317 records. Surprisingly, this doesn't seem to be tied with the installation date of the solar station, as it was built back in 2017 at a similar time to other solar stations like *Manchester Building M*.

Moving on to data preparation for our data visualizations, we've noticed that the `kWh` column is updated every hour according to the `date` column. To make assessing energy production more practical, we'll aggregate the data to obtain the total kWh production per day for each solar station. This daily summary will provide a more comprehensive view of energy generation trends, making it easier to identify patterns and variations later in the notebook.

In [41]:
solar_data['date'] = pd.to_datetime(solar_data['date'])
solar_data['day'] = solar_data['date'].dt.date

agg_funcs = {
    'kWh': 'sum',
    'installationDate': 'first'
}

summed_data = solar_data.groupby(['name', 'day']).agg(agg_funcs).reset_index()
summed_data.rename(columns={'day': 'date'}, inplace=True)
summed_data

name        date       kWh installationDate
0      Bearspaw Water Treatment Plant  2017-11-21   171.789       2017/10/31
1      Bearspaw Water Treatment Plant  2017-11-22  2582.676       2017/10/31
2      Bearspaw Water Treatment Plant  2017-11-23    59.662       2017/10/31
3      Bearspaw Water Treatment Plant  2017-11-24   130.120       2017/10/31
4      Bearspaw Water Treatment Plant  2017-11-28   360.454       2017/10/31
...                               ...         ...       ...              ...
22540  Whitehorn Multi-Service Centre  2023-08-25  1939.023       2016/12/13
22541  Whitehorn Multi-Service Centre  2023-08-26  2060.412       2016/12/13
22542  Whitehorn Multi-Service Centre  2023-08-27  1996.615       2016/12/13
22543  Whitehorn Multi-Service Centre  2023-08-28  1976.131       2016/12/13
22544  Whitehorn Multi-Service Centre  2023-08-29  2075.915       2016/12/13

[22545 rows x 4 columns]

Now that we've aggregated our data to a suitable format for visualization, let's compare different energy stations and visualize any similarities or notable differences in their kWh production per day. We can compare different stations by using *variables* in the code cell below. Think of a variable as a container that holds information. The variables's (**station1** and **station2**) information can be changed to any solar station in the dataframe. The list of stations can be found in our code cells above.

In [57]:
# The information in station1 and station2 can be changed by altering the information to the right of the variable
# For example, you can input CFD Firehall #7 instead of Southland Leisure Centre

station1 = 'Southland Leisure Centre'
station2 = 'Whitehorn Multi-Service Centre'

# List of allowed station names
allowed_stations = [
    'Southland Leisure Centre',
    'Whitehorn Multi-Service Centre',
    'City of Calgary North Corporate Warehouse',
    'Richmond - Knob Hill Community Hall',
    'Glenmore Water Treatment Plant',
    'Calgary Fire Hall Headquarters',
    'Bearspaw Water Treatment Plant',
    'CFD Firehall #7',
    'Manchester Building M',
    'Hillhurst Sunnyside Community Association',
    'Telus Spark'
]

if station1 in allowed_stations and station2 in allowed_stations:
    station_fig1 = make_subplots(rows=1, cols=2, subplot_titles=(f'kWH produced {station1} per Day', f'kWh produced by {station1} per Day'))
    station_fig1.add_trace(
        go.Scatter(
            x=summed_data[summed_data['name'] == station1]['date'],
            y=summed_data[summed_data['name'] == station1]['kWh'],
            name=f'kWh/day produced by {station1}'
        ),
        row=1, col=1
    )

    station_fig1.add_trace(
        go.Scatter(
            x=summed_data[summed_data['name'] == station2]['date'],
            y=summed_data[summed_data['name'] == station2]['kWh'],
            name=f'kWh/day produced by {station2}'
        ),
        row=1, col=2   
    )

    station_fig1.update_xaxes(title_text='Date')
    station_fig1.update_yaxes(title_text='kWh/day')
    station_fig1.update_layout(showlegend=False)
    station_fig1.show()
else:
    print("Invalid station names. Please use station names from the allowed list.")

Looking at the visualization, an immediate seasonal pattern emerges. Months such as `May`, `June`, `July`, and `August` all highlight the peak of kWh production regardless of which station is visualized. This can be attributed to a combination of factors. During these summer months, the Earth's tilt positions the sun more directly overhead, allowing solar panels to receive a higher intensity of sunlight. This is further enhanced by longer daylight hours, typical of the summer season, providing more time for panels to capture energy. Clearer skies and reduced cloud cover in summer also contribute to consistent and intense sunlight. 

On the flip-side, months such as `November`, `December`, `January`, and `February` exhibit very low kWh production. This dip in production during the winter months is primarily attributable to a range of factors that operate in contrast to those favoring high production in the summer. 

Firstly, during the winter season, the Earth's tilt places the sun at a lower angle in the sky, resulting in less direct sunlight reaching solar panels. The reduced angle spreads the same amount of solar energy over a larger surface area, which leads to diminished energy capture efficiency. Additionally, winter months are characterized by shorter daylight hours, which limit the window of opportunity for solar panels to generate electricity.

Secondly, winter weather conditions often introduce challenges for solar energy generation. In Calgary, snowfall is extremely prevalent and coupled with increased cloud cover, the amount of sunlight reaching solar panels is immensely reduced. 

Now that we've identified the seasonal factors that affect solar energy, let's find the total kWh production across all our solar panels to:

1. identify any trends based on years 
2. see if the total output of energy is significant

In [43]:
solar_data['date'] = pd.to_datetime(solar_data['date'])
solar_data['year'] = solar_data['date'].dt.year

# Find total kWh produced for each year
sum_kWh_per_year = solar_data.groupby('year')['kWh'].sum().reset_index()

year_fig = px.line(sum_kWh_per_year, x='year', y='kWh', title='Sum kWh Production per Year',
                   labels={'year': 'Year', 'kWh': 'Sum kWh Production'})

year_fig.update_traces(line=dict(color='red', width=3)).show() 

Looking at the figure, it appears that sum kWh production across different years seems to have plateaued across 2018-2020. We can also assume the dip in 2023 is due to the solar dataframe only containing information up to September 2023. 

Looking at the total kWh production, the years 2018, 2019, and 2020 has produced 1.7-1.76 million kWh of energy annually. This is a staggering number considering that the average household in Alberta uses [7,200 kWh of energy](https://energyrates.ca/residential-electricity-natural-gas/) per year. An estimate of 235-245 homes can be completely fueled on a yearly basis on energy that is constant and renewable, which is pretty crazy to think about!

We can create a similar visualization but include months alongside years to get a more detailed perspective on the cyclical solar energy pattern in Calgary. 

In [67]:
solar_data['date'] = pd.to_datetime(solar_data['date'])

solar_data['year'] = solar_data['date'].dt.year
solar_data['month'] = solar_data['date'].dt.month

# Find total kWh produced for each year and month
sum_kWh_per_month = solar_data.groupby(['year', 'month'])['kWh'].sum().reset_index()
sum_kWh_per_month['year_month'] = sum_kWh_per_month.apply(lambda x: f"{int(x['year'])}-{int(x['month']):02d}", axis=1)

month_fig = px.line(sum_kWh_per_month, x='year_month', y='kWh', title='Sum kWh Production per Year-Month',
                    labels={'year_month': 'Year-Month', 'kWh': 'Sum kWh Production'})

month_fig.update_traces(line=dict(color='green', width=3)).show()

A similar seasonal pattern emerges for each year in our visualization. In particular, July 2018 had the highest total kWh production at a staggering 283,675 kWh produced in a single month. This high is similarly matched in other years such as July 2019 (254,242 kWh), August 2020 (272,659 kWh), and June 2021 (231,633 kWh).

We can also utilize another column in our dataframe, `installationDate`, to see if there is a correlation between energy production and the date that particular station has built. Specifically, we want to see if older solar panels produce less or more energy compared to younger solar panels. 

In [68]:
summed_data['installationDate'] = pd.to_datetime(summed_data['installationDate'])

mean_avg_kWh = summed_data.groupby(['name', 'installationDate'])['kWh'].agg(['mean']).reset_index()

installation_fig = px.scatter(mean_avg_kWh, x='installationDate', y='mean', title='Installation Date vs Average kWh/day',
             text='mean', hover_data=['installationDate'], color='name', labels={'mean': 'Avg kWh/day'})

installation_fig.update_traces(texttemplate='%{text:.2f}')
installation_fig.update_layout(xaxis_title='Solar Panel Station',yaxis_title='Average kWh/day').show()

Looking at the visualization output, there doesn't appear to be any significant correlation of older machines producing less or more energy compared to younger machines. This is evidenced by various energy stations (ex. Bearspaw Water Treatment Plant vs Manchester Building M) having similar installation dates but having drastically different energy outputs (1572.78 kWh/day vs 100.54 kwh/day). 

In the final visualization in our notebook, we can represent each station's output in a different format, such as a map. This involves converting the addresses of our solar stations into latitude and longitude coordinates and then displaying them on a *Folium* map. Utilizing a map perspective offers valuable insights by visually presenting the geographical distribution of solar stations across Calgary, allowing us to easily observe their locations and energy production in a spatial context.

In [47]:
lat_long = [(51.016540, -114.036710),
            (51.085020, -113.982750),
            (50.961580, -114.107720),
            (51.057310, -114.092520),
            (51.002600, -114.099730),
            (51.072990, -114.005940),
            (51.029960, -114.116740),
            (51.076270, -114.071260),
            (51.028870, -114.046510),
            (51.054010, -114.024590),
            (51.104070, -114.253010)
]

combined_df = pd.DataFrame({'name': names, 'address': addresses, 'lat_long': lat_long})
average_kWh_per_name = summed_data.groupby('name')['kWh'].mean().reset_index()


combined_df = combined_df.merge(average_kWh_per_name, on='name', how='left')

combined_df = combined_df.rename(columns={'kWh': 'average_kWh'})
combined_df

name                   address  \
0              Calgary Fire Hall Headquarters             1212 42 AV SE   
1              Whitehorn Multi-Service Centre             3705 35 ST NE   
2                    Southland Leisure Centre      2000 SOUTHLAND DR SW   
3   Hillhurst Sunnyside Community Association              1320 5 AV NW   
4              Glenmore Water Treatment Plant             1634 56 AV SW   
5   City of Calgary North Corporate Warehouse             2340 22 ST NE   
6         Richmond - Knob Hill Community Hall             2433 26 AV SW   
7                             CFD Firehall #7              2708 4 ST NW   
8                       Manchester Building M             651M 25 AV SE   
9                                 Telus Spark      220 ST GEORGES DR NE   
10             Bearspaw Water Treatment Plant  11444 Bearspaw Dam RD NW   

                  lat_long  average_kWh  
0   (51.01654, -114.03671)    54.292083  
1   (51.08502, -113.98275)  1216.496896  
2   (50.96158, -114.10772)   459.805034  
3   (51.05731, -114.09252)    95.736196  
4    (51.0026, -114.09973)   742.429015  
5   (51.07299, -114.00594)   368.080654  
6   (51.02996, -114.11674)    35.564016  
7   (51.07627, -114.07126)    95.300733  
8   (51.02887, -114.04651)   100.541811  
9   (51.05401, -114.02459)    34.219846  
10  (51.10407, -114.25301)  1572.779154

In [66]:
all_locations_map = folium.Map(location=[51.05, -114.0719])

for index, row in combined_df.iterrows():
    lat, lon = row['lat_long']
    address = row['address']
    avg_kWh = row['average_kWh']
    
    folium.CircleMarker(
        location=[lat, lon],
        radius=avg_kWh/100 + 1,  # Size based on average kWh
        color='blue', 
        fill=True,
        fill_color='blue',
        fill_opacity=0.7,
        popup=f"Name: {row['name']}<br>Address: {address}<br>Average kWh/day: {avg_kWh}"  
    ).add_to(all_locations_map)

all_locations_map

# Interpret

In this notebook, we embarked on a comprehensive exploration of solar energy production in Calgary. Our primary goal was to gain insights into energy trends among various solar stations, and we've achieved this through a series of data preparation and visualization steps.

**Data Cleaning and Aggregation**: We began by organizing and cleaning our data, ensuring it was accurate, complete, and ready for analysis. We then aggregated the data to provide a daily summary of kWh production for each solar station, making it more practical for assessing energy production trends.

**Visualization for Comparison**: To compare different energy stations effectively, we employed visualizations. We utilized variables to select specific solar station names for comparison. These visualizations allowed us to discern seasonal patterns, similarities, and differences in daily kWh production, providing valuable insights into energy station performance.

**Geospatial Visualization**: In the final step, we used latitude and longitude coordinates derived from station addresses to create a Folium map. This map provided a geographical perspective, showcasing the locations of solar stations across Calgary. It allowed us to visualize the distribution of these stations and their energy output in a spatial context.

## Reflect on what you see

After making your visualization the next step is to use the data and your visualization to answer the question. Look at and interact with the visualization above. When you hover your mouse over the plots, you’ll notice more information appears. You can also use the legend to make plots appear and disappear.

#### Think about the following questions.

* What do you notice about these graphs?
* What do you wonder about the data?
* What kind of inferences can you make based on this data?
* Is there another way to visualize this data that would change your inerpretation of the information? 


#### Use the fill-in-the-blank prompts to summarize your thoughts.
* "I used to think _______"
* "Now I think _______"
* "I wish I knew more about _______"
* "These data visualizations remind me of _______"
* "I really like _______"

# Communicate

If you have not yet done this, use the plot to answer our question on which timezones have the largest differences between sunrise and sunset times. Vice-versa, which areas have the smallest differences?

How can you communicate that information? What kind of product could you create to share that information with your school community and wider community?

Consider tagging Callysto on [Twitter](https://twitter.com/callysto_canada), [YouTube](https://www.youtube.com/Callysto), [TikTok](https://www.tiktok.com/@callysto_canada), [Facebook](https://www.facebook.com/callystocanada/), or [Linkedin](https://www.linkedin.com/company/callysto-canada/) if you decide to share your reflections or projects on social media.

[![Callysto.ca License](https://github.com/callysto/curriculum-notebooks/blob/master/callysto-notebook-banner-bottom.jpg?raw=true)](https://github.com/callysto/curriculum-notebooks/blob/master/LICENSE.md)